### PR TITLE
[Security] [Guard] Removed useless param annotations

### PR DESCRIPTION
**Pull request by thomasbisignani**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/32254#issuecomment-506719957
| License       | MIT
| Doc PR        | 

The PR removes useless `@param` annotation, it is linked with the https://github.com/symfony/symfony/pull/32254#issuecomment-506719957

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php [hunk 1]
```diff
@@ -36,9 +36,6 @@ public function supports(Request $request)
      * Shortcut to create a PostAuthenticationGuardToken for you, if you don't really
      * care about which authenticated token you're using.
      *
-     * @param UserInterface $user
-     * @param string        $providerKey
```
Comment (derrabus) [inline src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php]: I‘d keep `@param string $providerKey` as long as we don’t have the type-hint here.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

File: src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php]: Agreed, @thomasbisignani Can you revert the removal of `string` params?

Comment (thomasbisignani) [reply src/Symfony/Component/Security/Guard/AbstractGuardAuthenticator.php]: Done. Thx !

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @thomasbisignani.

**Pull request merged**